app/src/main/java/com/github/se/studybuddies/ui/chat/ChatScreen.kt

```diff
@@ -451,3 +467,394 @@ fun PrivateChatTitle(chat: Chat) {
+          )
+          Spacer(modifier = Modifier.height(16.dp))
+
+          Text(text = stringResource(R.string.poll_options))
+
+          LazyColumn(modifier = Modifier.fillMaxWidth().heightIn(max = 200.dp)) {
+            items(options.value) { option ->
+              OutlinedTextField(
+                  value = option,
+                  onValueChange = {
+                    options.value =
+                        options.value.toMutableList().apply {
+                          set(indexOf(option), it)
+                          removeAll { option -> option.isBlank() }
+                          if (lastOrNull()?.isNotBlank() == true) add("")
+                        }
+                  },
+                  modifier = Modifier.fillMaxWidth().testTag("add_poll_options_text_field"),
+                  textStyle = TextStyle(color = Black),
+                  singleLine = true,
+                  placeholder = { Text(stringResource(R.string.enter_poll_options)) },
+              )
+              Spacer(modifier = Modifier.height(8.dp))
+            }
+          }
+          Row(
+              modifier =
+                  Modifier.fillMaxWidth().padding(8.dp).testTag("add_poll_single_choice_row"),
+              verticalAlignment = Alignment.CenterVertically,
+              horizontalArrangement = Arrangement.SpaceBetween) {
+                Text(
+                    text = stringResource(R.string.single_choice),
+                    style = TextStyle(color = Black),
+                    modifier = Modifier.testTag("add_poll_single_choice_text"))
+                Switch(
+                    checked = singleChoice.value,
+                    onCheckedChange = { singleChoice.value = it },
+                    colors =
+                        SwitchDefaults.colors(
+                            checkedThumbColor = com.github.se.studybuddies.ui.theme.White,
+                            checkedTrackColor = Blue,
+                            uncheckedTrackColor = Color.LightGray))
+              }
+        }
+      },
+      button = {
+        val nonEmptyOptions = options.value.filter { it.isNotBlank() }
+        SaveButton(question.value.isNotBlank() && nonEmptyOptions.size >= 2) {
+          messageViewModel.sendPollMessage(
+              question.value, singleChoice.value, nonEmptyOptions.toList())
+          showAddPoll.value = false
+          question.value = ""
+          options.value = listOf("")
+          singleChoice.value = true
+        }
+      })
+}
+
+@Composable
+fun ShowAlertDialog(
+    modifier: Modifier = Modifier,
+    showDialog: MutableState<Boolean>,
+    onDismiss: () -> Unit,
+    title: @Composable () -> Unit,
+    content: @Composable () -> Unit,
+    button: @Composable () -> Unit = {},
+) {
+  if (showDialog.value) {
+    AlertDialog(
+        modifier = modifier,
+        onDismissRequest = onDismiss,
+        text = content,
+        title = title,
+        confirmButton = button)
+  }
+}
```

Comment:
Theses functions have been moved over there : https://github.com/Study-Buddies-SwEnt/SwEnt_Group_project/blob/main/app/src/main/java/com/github/se/studybuddies/ui/chat/utility/ChatOption.kt

It was to reduce the size of the file and also it's complexity.

Can you not put it back here and make you changes over there